<h3>Capstone Project - Car accident severity</h3>

<h4>INTRODUCTION:</h4> We will be using the example dataset given by SDOT Traffic Management Division to predict the car accident severity of Seattle. This city is a place where rainfall is experienced for an average of 18 days/month and it also have high level of humidity and it makes it a perfect place for studying car accident severity due to the weather. Seattle had set a goal that by 2030 they'll have 0% of fatal or serious injuries for car accidents. Although these numbers have decreased through the years it seems very unlikely that the city will reach this goal for 2030. So the purpose of this project Y is to predict the severity of this accidents due to several variables so that the city can take actions accordingly.

In [2]:
import pandas as pd
import numpy as np

In [3]:
dataset = pd.read_csv('https://s3.us.cloud-object-storage.appdomain.cloud/cf-courses-data/CognitiveClass/DP0701EN/version-2/Data-Collisions.csv')
dataset.shape

(194673, 38)

In [4]:
dataset.dtypes.count

<bound method Series.count of SEVERITYCODE        int64
X                 float64
Y                 float64
OBJECTID            int64
INCKEY              int64
COLDETKEY           int64
REPORTNO           object
STATUS             object
ADDRTYPE           object
INTKEY            float64
LOCATION           object
EXCEPTRSNCODE      object
EXCEPTRSNDESC      object
SEVERITYCODE.1      int64
SEVERITYDESC       object
COLLISIONTYPE      object
PERSONCOUNT         int64
PEDCOUNT            int64
PEDCYLCOUNT         int64
VEHCOUNT            int64
INCDATE            object
INCDTTM            object
JUNCTIONTYPE       object
SDOT_COLCODE        int64
SDOT_COLDESC       object
INATTENTIONIND     object
UNDERINFL          object
WEATHER            object
ROADCOND           object
LIGHTCOND          object
PEDROWNOTGRNT      object
SDOTCOLNUM        float64
SPEEDING           object
ST_COLCODE         object
ST_COLDESC         object
SEGLANEKEY          int64
CROSSWALKKEY        int64
HITPARKE

<h4>Data Description:</h4> We will be using the example data set given by SDOT Traffic Management Division that contains 194673 rows and 38 columns of data. This columns contains a vast amount of variables such as location, vehicle count, road condition and lightning condition, weather, collision type and many others. With these information we'll train the dataset to predict the accident severity due to most of this variables but most of it focused on the weather and road conditions. We'll need to separate this data set to a train dataset and a test dataset to have the best results possible.